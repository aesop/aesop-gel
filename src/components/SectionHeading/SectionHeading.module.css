@import 'styles/index';

.base {
  position: relative;
  display: flex;
  flex-direction: column;
}

.eyebrow {
  order: 1;
}

.heading {
  order: 2;
}

.subHeading {
  order: 3;
}

.offsetPageHeading {
  @media (--viewport-sm-only) {
    padding: 0 20px;
  }

  @media (--viewport-md) {
    width: 40%;
    margin-top: 85px;
    margin-left: 30%;
  }

  @media (--viewport-lg) {
    margin-top: 110px;
  }
}

.SuisseTitleFont {
  font-family: var(--font-family-sans-serif-regular);
}

.ZapfTitleFont {
  @extend %zaph-serif-typography;
}
