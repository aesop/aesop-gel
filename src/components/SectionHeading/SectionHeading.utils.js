export const getHeadingLevel = (eyebrow, isPageHeading) =>
  eyebrow ? (isPageHeading ? '2' : '3') : isPageHeading ? '1' : '2';

export const getSubHeadingLevel = (eyebrow, heading, isPageHeading) => {
  if (eyebrow) {
    if (heading) {
      return isPageHeading ? '3' : '4';
    }

    return isPageHeading ? '2' : '3';
  }

  if (heading) {
    return isPageHeading ? '2' : '3';
  }

  return isPageHeading ? '1' : '2';
};
