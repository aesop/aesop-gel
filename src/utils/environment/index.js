/* istanbul ignore file */
import { isInBrowser } from './environment';

export { isInBrowser };

export default { isInBrowser };
