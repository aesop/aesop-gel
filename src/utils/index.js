export * as environment from './environment';
export * as objects from './objects';
export * as product from './product';
export * as viewports from './viewports';
