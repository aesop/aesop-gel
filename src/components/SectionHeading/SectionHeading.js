import React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';
import Heading from '~/components/Heading';
import { getHeadingLevel, getSubHeadingLevel } from './SectionHeading.utils';
import styles from './SectionHeading.module.css';

const SectionHeading = ({
  childrenClassNames,
  className,
  eyebrow,
  hasSerifFontHeading,
  heading,
  id,
  isFlush,
  isOffsetPageHeading,
  isPageHeading,
  subHeading,
  theme,
  paddingBottom,
  fontFamily,
}) => {
  const classSet = cx(
    styles.base,
    { [styles.offsetPageHeading]: isOffsetPageHeading },
    className,
  );
  const eyebrowClassSet = cx(childrenClassNames.eyebrow);
  const headingClassSet = cx(
    childrenClassNames.heading,
    paddingBottom && styles[`${paddingBottom}PaddingBottom`],
    fontFamily && styles[`${fontFamily}FontFamily`],
  );
  const subHeadingClassSet = cx(childrenClassNames.subHeading);

  return (
    <header className={classSet} id={id}>
      {eyebrow && (
        <Heading
          className={eyebrowClassSet}
          hasMediumWeightFont={true}
          isFlush={isFlush}
          level={isPageHeading ? '1' : '2'}
          size="xXSmall"
          theme={theme}
        >
          {eyebrow}
        </Heading>
      )}
      {heading && (
        <Heading
          className={headingClassSet}
          hasSerifFont={hasSerifFontHeading}
          isFlush={isFlush}
          level={getHeadingLevel(eyebrow, isPageHeading)}
          size="xLarge"
          theme={theme}
        >
          {heading}
        </Heading>
      )}
      {subHeading && (
        <Heading
          className={subHeadingClassSet}
          hasMediumWeightFont={true}
          isFlush={isFlush}
          level={getSubHeadingLevel(eyebrow, heading, isPageHeading)}
          size="xSmall"
          theme={theme}
        >
          {subHeading}
        </Heading>
      )}
    </header>
  );
};

SectionHeading.propTypes = {
  childrenClassNames: PropTypes.shape({
    eyebrow: PropTypes.string,
    heading: PropTypes.string,
    subHeading: PropTypes.string,
  }),
  className: PropTypes.string,
  eyebrow: PropTypes.string,
  hasSerifFontHeading: PropTypes.bool,
  heading: PropTypes.string,
  id: PropTypes.string,
  isFlush: PropTypes.bool,
  isOffsetPageHeading: PropTypes.bool,
  isPageHeading: PropTypes.bool,
  subHeading: PropTypes.string,
  theme: PropTypes.oneOf(['dark', 'light']),
  paddingBottom: PropTypes.oneOf(['none', 'small', 'medium', 'large']),
  fontFamily: PropTypes.oneOf(['Suisse', 'Zapf']),
};

SectionHeading.defaultProps = {
  childrenClassNames: {
    eyebrow: undefined,
    heading: undefined,
    subHeading: undefined,
  },
  className: undefined,
  eyebrow: undefined,
  hasSerifFontHeading: false,
  heading: undefined,
  id: undefined,
  isFlush: false,
  isOffsetPageHeading: false,
  isPageHeading: false,
  subHeading: undefined,
  theme: 'dark',
  paddingBottom: undefined,
  fontFamily: undefined,
};

export default SectionHeading;
