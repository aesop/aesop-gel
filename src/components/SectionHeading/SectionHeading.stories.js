import React from 'react';
import { storiesOf } from '@storybook/react';
import { text, select, boolean } from '@storybook/addon-knobs';
import '~.storybook/storybook.module.css';
import SectionHeading from './SectionHeading';

storiesOf('Components/SectionHeading', module).add('Base component', () => (
  <SectionHeading
    eyebrow={text(
      'eyebrow',
      'Eyebrow: Offers fortification of the highest order against free radicals.',
    )}
    heading={text(
      'heading',
      'Heading: Fortification of the highest order. Parsley Seed Anti-Oxidant Facial Hydrating Cream.',
    )}
    id="Anti-Oxidant"
    isPageHeading={boolean('isPageHeading', false)}
    subHeading={text(
      'subHeading',
      'SubHeading: Parsley Seed Cream. Intensely soothing, nourishing and hydrating, this elegant formulation contains a potent blend of botanicals that offer fortification of the highest order against free radicals.',
    )}
    theme={select('theme', ['dark', 'light'], 'dark')}
  />
));
