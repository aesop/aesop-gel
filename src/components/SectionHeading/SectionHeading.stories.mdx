import { Meta, Story, Preview, Props } from '@storybook/addon-docs/blocks';
import * as knobs from '@storybook/addon-knobs';
import SectionHeading from './SectionHeading';

<Meta title="Components/SectionHeading" component={SectionHeading} />

# SectionHeading

<Preview>
  <Story name="Base component">
    <div style={{ border: '1px solid #333', margin: '20px'}}>
      <SectionHeading
        eyebrow={knobs.text(
          'eyebrow',
          'Eyebrow: Offers fortification of the highest order against free radicals.',
          'Copy',
        )}
        heading={knobs.text(
          'heading',
          'Heading: Fortification of the highest order. Parsley Seed Anti-Oxidant Facial Hydrating Cream.',
          'Copy',
        )}
        isFlush={knobs.boolean('isFlush', false, 'Presentation')}
        id="Anti-Oxidant"
        isPageHeading={knobs.boolean('isPageHeading', false, 'Presentation')}
        subHeading={knobs.text(
          'subHeading',
          'SubHeading: Parsley Seed Cream. Intensely soothing, nourishing and hydrating, this elegant formulation contains a potent blend of botanicals that offer fortification of the highest order against free radicals.',
          'Copy',
        )}
        theme={knobs.select('theme', ['dark', 'light'], 'dark', 'Presentation')}
        fontFamily={knobs.select('fontFamily', ['Suisse', 'Zapf'], 'Zapf', 'Presentation')}
        paddingBottom={knobs.select('paddingBottom', ['none', 'small', 'medium', 'large'], 'none', 'Presentation')}
      />
    </div>
    <p>SubHeading: Parsley Seed Cream. Intensely soothing, nourishing and hydrating, this elegant formulation cont</p>
  </Story>
</Preview>

## API

<Props of={SectionHeading} />

## Variations

### Page Heading

<Preview>
  <Story name="Page Heading">
    <SectionHeading
      heading="Page Heading: Fortification of the highest order. Parsley Seed Anti-Oxidant Facial Hydrating Cream."
      isPageHeading={true}
    />
  </Story>
</Preview>
