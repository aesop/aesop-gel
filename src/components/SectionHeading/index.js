/* istanbul ignore file */
import { SectionHeading } from './SectionHeading';

export { SectionHeading };
