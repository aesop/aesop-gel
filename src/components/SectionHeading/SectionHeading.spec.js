import React from 'react';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import { getHeadingLevel, getSubHeadingLevel } from './SectionHeading.utils';
import SectionHeading from './SectionHeading';

configure({ adapter: new Adapter() });

describe('<SectionHeading />', () => {
  it('should be defined', () => {
    expect(SectionHeading).toBeDefined();
  });

  it('renders base component correctly', () => {
    const tree = renderer
      .create(
        <SectionHeading
          heading="Title: Fortification of the highest order. Parsley Seed Anti-Oxidant Facial Hydrating Cream."
          id="Anti-Oxidant"
        />,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders with subHeading variation correctly', () => {
    const tree = renderer
      .create(
        <SectionHeading
          heading="Heading: Fortification of the highest order."
          id="Parsley"
          subHeading="Subcopy: Parsley Seed Anti-Oxidant Facial Hydrating Cream 60mL"
        />,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders with subHeading and eyebrow variation correctly', () => {
    const tree = renderer
      .create(
        <SectionHeading
          eyebrow="Eyebrow: Offers fortification of the highest order against free radicals."
          heading="Heading: Fortification of the highest order. Parsley Seed Anti-Oxidant Facial Hydrating Cream."
          id="Anti-Oxidant"
          subHeading="SubHeading: Parsley Seed Cream. Intensely soothing, nourishing and hydrating, this elegant formulation contains a potent blend of botanicals that offer fortification of the highest order against free radicals."
        />,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders with subHeading, eyebrow variation, font family correctly', () => {
    const tree = renderer
      .create(
        <SectionHeading
          eyebrow="Eyebrow: Offers fortification of the highest order against free radicals."
          heading="Heading: Fortification of the highest order. Parsley Seed Anti-Oxidant Facial Hydrating Cream."
          id="Anti-Oxidant"
          subHeading="SubHeading: Parsley Seed Cream. Intensely soothing, nourishing and hydrating, this elegant formulation contains a potent blend of botanicals that offer fortification of the highest order against free radicals."
          titleFont="Zapf"
        />,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});

describe('SectionHeading.utils', () => {
  it('renders heading as an `h2` if no `eyebrow` prop is passed', () => {
    expect(getHeadingLevel(undefined)).toEqual('2');
  });

  it('renders heading as an `h3` if `eyebrow` prop is passed', () => {
    expect(getHeadingLevel('eyebrow')).toEqual('3');
  });

  it('renders subHeading as an `h2` if no `eyebrow` or `heading` prop are passed', () => {
    expect(getSubHeadingLevel(undefined, undefined)).toEqual('2');
  });

  it('renders subHeading as an `h3` if `eyebrow` prop is passed but no `heading` prop is passed', () => {
    expect(getSubHeadingLevel('eyebrow', undefined)).toEqual('3');
  });

  it('renders subHeading as an `h3` if no `eyebrow` prop is passed but `heading` prop is passed', () => {
    expect(getSubHeadingLevel(undefined, 'subHeading')).toEqual('3');
  });

  it('renders subHeading as an `h4` if `eyebrow` and `heading` props are passed', () => {
    expect(getSubHeadingLevel('eyebrow', 'subHeading')).toEqual('4');
  });
});
