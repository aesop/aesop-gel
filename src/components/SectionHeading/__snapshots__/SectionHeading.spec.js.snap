// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SectionHeading /> renders base component correctly 1`] = `
<header
  className="base"
  id="Anti-Oxidant"
>
  <h2
    className="base xLarge dark"
  >
    Title: Fortification of the highest order. Parsley Seed Anti-Oxidant Facial Hydrating Cream.
  </h2>
</header>
`;

exports[`<SectionHeading /> renders with subHeading and eyebrow variation correctly 1`] = `
<header
  className="base"
  id="Anti-Oxidant"
>
  <h2
    className="base mediumWeightFont xXSmall dark"
  >
    Eyebrow: Offers fortification of the highest order against free radicals.
  </h2>
  <h3
    className="base xLarge dark"
  >
    Heading: Fortification of the highest order. Parsley Seed Anti-Oxidant Facial Hydrating Cream.
  </h3>
  <h4
    className="base mediumWeightFont xSmall dark"
  >
    SubHeading: Parsley Seed Cream. Intensely soothing, nourishing and hydrating, this elegant formulation contains a potent blend of botanicals that offer fortification of the highest order against free radicals.
  </h4>
</header>
`;

exports[`<SectionHeading /> renders with subHeading variation correctly 1`] = `
<header
  className="base"
  id="Parsley"
>
  <h2
    className="base xLarge dark"
  >
    Heading: Fortification of the highest order.
  </h2>
  <h3
    className="base mediumWeightFont xSmall dark"
  >
    Subcopy: Parsley Seed Anti-Oxidant Facial Hydrating Cream 60mL
  </h3>
</header>
`;

exports[`<SectionHeading /> renders with subHeading, eyebrow variation, font family correctly 1`] = `
<header
  className="base"
  id="Anti-Oxidant"
>
  <h2
    className="base mediumWeightFont xXSmall dark"
  >
    Eyebrow: Offers fortification of the highest order against free radicals.
  </h2>
  <h3
    className="base xLarge dark ZapfTitleFont"
  >
    Heading: Fortification of the highest order. Parsley Seed Anti-Oxidant Facial Hydrating Cream.
  </h3>
  <h4
    className="base mediumWeightFont xSmall dark"
  >
    SubHeading: Parsley Seed Cream. Intensely soothing, nourishing and hydrating, this elegant formulation contains a potent blend of botanicals that offer fortification of the highest order against free radicals.
  </h4>
</header>
`;
